### PR TITLE
Turn on/off classification augmentations

### PR DESCRIPTION
**Pull request by goodsong81**
### Summary

* Add 'enable' flag for classification recipe transforms (default=True)
* Add common augmentations (disabled by default for backward compatibility)
```python
configurable_augs = [
    "PhotoMetricDistortion",
    "RandomAffine",
    "RandomVerticalFlip",
    "GaussianBlur",
    "GaussianNoise",
]
```
* Add aug on/off combination tests for all classification recipes

### How to test

```console
pytest -vs tests/integration/api/test_augmentation.py
```

### Checklist

- [x] I have added unit tests to cover my changes.​
- [x] I have added integration tests to cover my changes.​
- [ ] I have ran e2e tests and there is no issues.
- [x] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [x] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by eunwoosh — COMMENTED** [review]
LGTM

**Review by eunwoosh — APPROVED** [review]

**Review by harimkang — APPROVED** [review]

File: src/otx/recipe/classification/multi_class_cls/efficientnet_b0.yaml [hunk 1]
```diff
@@ -42,10 +42,20 @@ overrides:
         - class_path: otx.core.data.transform_libs.torchvision.EfficientNetRandomCrop
           init_args:
             scale: $(input_size)
+        - class_path: otx.core.data.transform_libs.torchvision.PhotoMetricDistortion
+          enable: false
+        - class_path: otx.core.data.transform_libs.torchvision.RandomAffine
+          enable: false
         - class_path: otx.core.data.transform_libs.torchvision.RandomFlip
```
Comment (kprokofi) [inline src/otx/recipe/classification/multi_class_cls/efficientnet_b0.yaml:49]: Why do we need to keep both RandomFlip (horizontal) and RandomVerticalFlip? I think horizontal flip is more common. Will these augmentations also be used for other tasks?

File: src/otx/recipe/classification/multi_class_cls/efficientnet_b0.yaml [hunk 2]
```diff
@@ -42,10 +42,20 @@ overrides:
...
           init_args:
             prob: 0.5
             is_numpy_to_tvtensor: true
+        - class_path: torchvision.transforms.v2.RandomVerticalFlip
+          enable: false
+        - class_path: torchvision.transforms.v2.GaussianBlur
+          enable: false
+          init_args:
+            kernel_size: 5
```
Comment (kprokofi) [inline src/otx/recipe/classification/multi_class_cls/efficientnet_b0.yaml:58]: This feature for now will not have possibility to change parameters, right?

**Review by kprokofi — COMMENTED** [review]
Can we discuss common list of augmentations? Do we have a requirement about the amount of augmentations to provide for Geti?

File: src/otx/recipe/classification/multi_class_cls/efficientnet_b0.yaml (hunk 2 above)
Comment (goodsong81) [reply src/otx/recipe/classification/multi_class_cls/efficientnet_b0.yaml:58]: Yes, right. But in this phase, it's a bit limited due to the UX.

File: src/otx/recipe/classification/multi_class_cls/efficientnet_b0.yaml (hunk 1 above)
Comment (goodsong81) [reply src/otx/recipe/classification/multi_class_cls/efficientnet_b0.yaml:49]: No requirement for augmentation list, but I've collected some from other solutions.
image

**Comment by goodsong81** [comment]
> Can we discuss common list of augmentations? Do we have a requirement about the amount of augmentations to provide for Geti?

Let's review the augmentation list after we have some working model. Currently, next series of PRs might have no impact for OTX as they will be turned on/off by default according to the current settings.

**Pull request merged**